spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTypeConverter.java

```diff
@@ -43,6 +79,33 @@ public static DataType toSparkType(Type gravitinoType) {
       return DataTypes.FloatType;
     } else if (gravitinoType instanceof Types.DoubleType) {
       return DataTypes.DoubleType;
+    } else if (gravitinoType instanceof Types.ByteType) {
+      return DataTypes.ByteType;
+    } else if (gravitinoType instanceof Types.ShortType) {
+      return DataTypes.ShortType;
+    } else if (gravitinoType instanceof Types.LongType) {
+      return DataTypes.LongType;
+    } else if (gravitinoType instanceof Types.DecimalType) {
+      Types.DecimalType decimalType = (Types.DecimalType) gravitinoType;
+      return DataTypes.createDecimalType(decimalType.precision(), decimalType.scale());
+    } else if (gravitinoType instanceof Types.BinaryType) {
+      return DataTypes.BinaryType;
+    } else if (gravitinoType instanceof Types.FixedCharType) {
+      Types.FixedCharType charType = (Types.FixedCharType) gravitinoType;
+      return CharType.apply((charType.length()));
+    } else if (gravitinoType instanceof Types.VarCharType) {
+      Types.VarCharType varCharType = (Types.VarCharType) gravitinoType;
+      return VarcharType.apply(varCharType.length());
+    } else if (gravitinoType instanceof Types.DateType) {
+      return DataTypes.DateType;
+    } else if (gravitinoType instanceof Types.TimestampType
+        && ((Types.TimestampType) gravitinoType).hasTimeZone()) {
+      return DataTypes.TimestampType;
+    } else if (gravitinoType instanceof Types.TimestampType
+        && !((Types.TimestampType) gravitinoType).hasTimeZone()) {
+      return DataTypes.TimestampNTZType;
+    } else if (gravitinoType instanceof Types.NullType) {
+      return DataTypes.NullType;
     }
```

Comment:
do you want to support Internal types in this PR?

Reply:
We can actually remove it from this pr, it's exactly not a primitive type. So should we put it with `userDdefinedType` or something? But `userDdefinedType` is not supported by gravitino yet

Reply:
Or we'll support it later.

Reply:
ok, let's support it latter